state/tests/Test_StateMask.pf

```diff
@@ -0,0 +1,93 @@
+#include "MAPL_Generic.h"
+
+module Test_StateMask
+
+   use state_utils_setup
+   use ESMF
+   use pfunit
+   use MAPL_ExceptionHandling
+   use MAPL_StateUtils
+
+   implicit none
+
+contains
+
+   @Before
+   subroutine set_up_data(this)
+      class(MpiTestMethod), intent(inout) :: this
+ 
+      integer :: status, rc
+
+      grid = ESMF_GridCreateNoPeriDim(countsPerDeDim1=[3], countsPerDeDim2=[3], _RC)
+      field_2d = ESMF_FieldCreate(grid, ESMF_TYPEKIND_R4, name="field_2d", _RC)
+      field_3d = ESMF_FieldCreate(grid, ESMF_TYPEKIND_R4, name="field_3d",ungriddedLBound=[1],ungriddedUBound=[2], _RC)
+      mask_field = ESMF_FieldCreate(grid, ESMF_TYPEKIND_R4, name="region_mask", _RC)
+		state = ESMF_StateCreate(fieldList=[field_2d,field_3d,mask_field], _RC)
+
+   end subroutine set_up_data
+
+   @after
+   subroutine teardown(this)
+      class(MpiTestMethod), intent(inout) :: this
+      call ESMF_FieldDestroy(field_2d, noGarbage=.true.)
+      call ESMF_FieldDestroy(field_3d, noGarbage=.true.)
+      call ESMF_FieldDestroy(mask_field, noGarbage=.true.)
+      call ESMF_StateDestroy(state, noGarbage=.true.)
+   end subroutine teardown
+
+   @Test(npes=[1])
```

Comment:
I think you need to make all of these ESMF_TestMethods to be safe.    If someone runs this suite on 4 cores, ESMF will try to spread these things across multiple processors.

```suggestion
   @test(type=ESMF_TestMethod, npes=[1])
```